x/stakeibc/keeper/msg_server_instant_redeem_stake_test.go

```diff
@@ -0,0 +1,201 @@
+		Status:             recordtypes.DepositRecord_TRANSFER_QUEUE,
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx, epochTracker)
+	s.App.RecordsKeeper.SetDepositRecord(s.Ctx, oldDepositRecord)
+	s.App.RecordsKeeper.SetDepositRecord(s.Ctx, initialDepositRecord)
+
+	return InstantRedeemStakeTestCase{
+		user:        user,
+		zoneAccount: zoneAccount,
+		initialState: InstantRedeemStakeState{
+			epochNumber:         epochTracker.EpochNumber,
+			depositRecordAmount: depositAmount,
+			initialAmount:       initialAmount,
+			hostZone:            hostZone,
+		},
+		validMsg: stakeibctypes.MsgInstantRedeemStake{
+			Creator:  user.acc.String(),
+			HostZone: HostChainId,
+			Amount:   unbondAmount,
+		},
+	}
+}
+
+// TODO: Need to add tests for at least multiple deposit records, non 1.0 Redemption Rates, and probably some other basic scenarios.
+func (s *KeeperTestSuite) TestInstantRedeemStake_Successful() {
+	tc := s.SetupInstantRedeemStake()
+	user := tc.user
+	zoneAccount := tc.zoneAccount
+	initialStAtomSupply := s.App.BankKeeper.GetSupply(s.Ctx, StAtom)
+	msg := tc.validMsg
+
+	// Validate Instant Redeem Stake
+	_, err := s.GetMsgServer().InstantRedeemStake(sdk.WrapSDKContext(s.Ctx), &msg)
+	s.Require().NoError(err)
+
+	// User STUATOM balance should have DECREASED by the amount unbonded
+	expectedUserStAtomBalance := user.stAtomBalance.SubAmount(msg.Amount)
+	actualUserStAtomBalance := s.App.BankKeeper.GetBalance(s.Ctx, user.acc, StAtom)
+	s.CompareCoins(expectedUserStAtomBalance, actualUserStAtomBalance, "user stuatom balance")
+	// User IBC/UATOM balance should have INCREASED by the amount unbonded
+	expectedUserAtomBalance := user.atomBalance.AddAmount(msg.Amount)
+	actualUserAtomBalance := s.App.BankKeeper.GetBalance(s.Ctx, user.acc, IbcAtom)
+	s.CompareCoins(expectedUserAtomBalance, actualUserAtomBalance, "user ibc/uatom balance")
+	// zoneAccount IBC/UATOM balance should have DECREASED by the size of the stake
+	expectedzoneAccountAtomBalance := zoneAccount.atomBalance.SubAmount(msg.Amount)
+	actualzoneAccountAtomBalance := s.App.BankKeeper.GetBalance(s.Ctx, zoneAccount.acc, IbcAtom)
+	s.CompareCoins(expectedzoneAccountAtomBalance, actualzoneAccountAtomBalance, "zoneAccount ibc/uatom balance")
+	// Bank supply of STUATOM should have DECREASED by the size of the stake
+	expectedBankSupply := initialStAtomSupply.SubAmount(msg.Amount)
+	actualBankSupply := s.App.BankKeeper.GetSupply(s.Ctx, StAtom)
+	s.CompareCoins(expectedBankSupply, actualBankSupply, "bank stuatom supply")
+
+	// Validate Instant Redeem Stake, subsequent call will not have funds.
+	_, err = s.GetMsgServer().InstantRedeemStake(sdk.WrapSDKContext(s.Ctx), &msg)
+	s.Require().EqualError(err, fmt.Sprintf("balance is lower than redemption amount. redemption amount: %v, balance %v: : invalid coins", msg.Amount, actualUserStAtomBalance.Amount))
+}
+
+func (s *KeeperTestSuite) TestInstantRedeemStake_InvalidCreatorAddress() {
```

Comment:
For future reference, these tests are mostly covered by the ValidateBasic unit test, so you really only one of these test cases to cover the error path in MsgInstantRedeem.

That said, no need to remove these cause there's nothing wrong with more tests!